CLI, update yaml, calc mm_score, wall_list bug

hwchen opened the pull request:
- add Tournament.calculate_mm_score()
- modify Tournament.wall_list() to check if round completed
- new file mm_cli.py. CLI flags for filename, new_round, pairings_list,
  wall_list, add_result
- modify tournament.yaml to include python constructors
- add tournament-init.yaml for easy tournament restart for testing

hwchen committed:
CLI, update yaml, calc mm_score, wall_list bug

- add Tournament.calculate_mm_score()
- modify Tournament.wall_list() to check if round completed
- new file mm_cli.py. CLI flags for filename, new_round, pairings_list,
    wall_list, add_result
- modify tournament.yaml to include python constructors
- add tournament-init.yaml for easy tournament restart for testing

hwchen committed:
Update wall_list

- update wall_list to return a str instead of printing inside function.

hndrewaall commented on mm_cli.py:
```diff
@@ -0,0 +1,63 @@
+                    action="store",
+                    default="tournament.yaml",
+                    help="Filename of data. Default is 'tournament.yaml'")
+parser.add_argument('--new-round', '-n',
```
Personally I would prefer to avoid dashes for the actual commands (as opposed to flags like filename which merely modify said commands). So, for example, you might write `mm-cli new-round`, or `mm-cli -f  /tmp/bs.yaml new-round`. Basically, I see the command statements as arguments, not flags.

hndrewaall commented on mm_cli.py:
```diff
@@ -0,0 +1,63 @@
+parser.add_argument('--new-round', '-n',
+                    action="store_true",
+                    help="Generate new round")
+parser.add_argument('--pairings_list', '-p',
```
Death to underscores

hndrewaall commented on mm_cli.py:
```diff
@@ -0,0 +1,63 @@
+
+import argparse
+import mcmahon
+import yaml
```
See PEP8 on organizing imports. Should be builtins, newline, thirdparty, newline, project/local. So in this case, argparse, newline, yaml, newline, mcmahon. I am the king of nitpicks

hndrewaall replied:
Also, two lines between imports and body. Get a friggin linter!

hndrewaall commented on mm_cli.py:
```diff
@@ -0,0 +1,63 @@
+    h = open(args.filename, 'w')
+    h.write(yaml.dump(tournament))
+    h.close()
+
```
Main here is implicit. Should at the very least be inside an `"if __name__ == '__main__'"`

hndrewaall replied:
Do we also want a command to start a new tournament? Otherwise we have to have an existing valid YAML file for it to work.

hndrewaall commented on mcmahon.py:
```diff
@@ -86,6 +86,17 @@ def new_tournament(cls, players=None):
+            if self.round_is_finished(idx):
+                for board in round_.values():
+                    self.players[board.winner].mm_score += 1
+
```
We should add a unit test for this

hndrewaall commented on tournament-init.yaml:
```diff
@@ -0,0 +1,12 @@
+  5: !player {aga_id: 4, name: Neil, rank: 2, mm_score: 0}
+rounds: []
+old_pairs: !!set {}
+current_players: !!set {}
```
I don't quite understand the purpose of this file.

hwchen replied:
I guess we don't need it in the repo. Using the cli overwrites the tournament.yaml file, so I kept this file on hand for an easy re-init.

hndrewaall replied:
Can also `git checkout -- tournament.yaml` to reset it to what it's committed as.

hndrewaall commented on mcmahon.py:
```diff
@@ -219,24 +230,26 @@ def wall_list(self):
-            print('{}: {} {}'.format(player_obj.name, player_obj.mm_score, 
-                                     str(wall_dict[player])))
-
+        for idx, round_ in enumerate(self.rounds):
```
Fancy! You just taught me a Python

hwchen committed:
CLI, whitespace fixes

- refactored mm_cli into a class-based arg parsing system. Enables commands with no flags, e.g. 'git fetch'
- whitespace fixes in mcmahon.py
